specs/agents/metadata.md

```diff
@@ -108,7 +107,9 @@ On Linux, the container ID and some of the Kubernetes metadata can be extracted
 
  If we match, then the basename is assumed to be a container ID.
 
-If the Kubernetes pod name is not the hostname, it can be overridden by the `KUBERNETES_POD_NAME` environment variable, using the [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/). In a similar manner, you can inform the agent of the node name and namespace, using the environment variables `KUBERNETES_NODE_NAME` and `KUBERNETES_NAMESPACE`.
+Sometimes the `KUBERNETES_POD_NAME` is set using the [Downward API](https://kubernetes.io/docs/tasks/inject-data-application/environment-variable-expose-pod-information/),
+so we set the pod name to its value if it exists.
+In a similar manner, you can inform the agent of the node name and namespace, using the environment variables `KUBERNETES_NODE_NAME` and `KUBERNETES_NAMESPACE`.
```

Comment:
nit: Perhaps also mention `KUBERNETES_POD_UID` to set/override `metadata.system.kubernetes.pid.uid`? Though that's adding scope. :)

Reply:
I added the `KUBERNETES_POD_UID` to the spec. I don't know why it wasn't mentioned, unless I'm missing some special logic?

Reply:
> I don't know why it wasn't mentioned, unless I'm missing some special logic?

I think it was just overlooked when added. AFAICT all the agents implement reading `KUBERNETES_POD_UID`.